lib: mpsl: Add the MPSL library

sofu1 opened the pull request:
Add the obfuscated library files, and the public header files.
Add references to MPSL lib and folders to CMakeLists and Kconfig
files so that MPSL can be linked in.
For now hardcoded to only link the soft version in mpsl/CMakeLists.

Signed-off-by: Solveig Fure <solveig.fure@nordicsemi.no>

tejlmand committed:
nrf_security: Updated documentation for nRF Security module

- Documenting the advanced configuration section in nRF Security.
- Improved dependencies on mbed TLS advanced configuration sections for
  improved user experience.
- Fixed api inclusion of nrf_cc310_platform APIs

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
Signed-off-by: Bartosz Gentkowski <bartosz.gentkowski@nordicsemi.no>

sofu1 committed:
lib: mpsl: Add the MPSL library

Add the obfuscated library files, and the public header files.
Add references to MPSL lib and folders to CMakeLists and Kconfig files
so that MPSL can be linked in.
Add Kconfig select for MULTITHREADING_LOCK for BLE Controller as its
glue layers and subsys in fw-nrfconnect-nrf depend on it.

Signed-off-by: Solveig Fure <solveig.fure@nordicsemi.no>

tejlmand committed:
doxygen: Added missing * in API description

Added a missing * in nrf_cc310_platform_header.h

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
crypto: Updated nrf_oberon to version 3.0.2

Adding nrf_oberon v3.0.2
Removing nrf_oberon v3.0.0

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

rugeGerritsen commented on CMakeLists.txt:
```diff
@@ -7,6 +7,7 @@
 set(NRFXLIB_DIR ${CMAKE_CURRENT_LIST_DIR} CACHE PATH "NRFXLIB root directory")
 add_subdirectory_ifdef(CONFIG_NRFXLIB_NFC             nfc)
 add_subdirectory_ifdef(CONFIG_BT_LL_NRFXLIB           ble_controller)
+add_subdirectory_ifdef(CONFIG_MPSL_NRFXLIB            mpsl)
```
There will not be another MPSL. Maybe we should name it `MPSL_LIBRARY` ?

sofu1 replied:
Done

rugeGerritsen commented on mpsl/README.rst:
```diff
@@ -0,0 +1,21 @@
+====
+
+Multi-Protocol Service Layer
+
```
Add some introduction text here describing the main purpose and functionality of the library?

sofu1 replied:
Can add something like this for now: "Multi-Protocol Service Layer (MPSL) is a library of common services for single and multi-protocol wireless implementations."
But we should add more stuff here later.

sofu1 replied:
Done

rugeGerritsen commented on mpsl/doc/mpsl.rst:
```diff
@@ -0,0 +1,6 @@
+How to use
+==========
+
+TODO
```
TODO

sofu1 replied:
I'll remove this

sofu1 replied:
Added a brief explanation

rugeGerritsen committed:
bluetooth: controller: Add SOC_NRF52811 to supported platforms

More platforms are supported, but these are not yet supported
by NCS. See README.rst

Signed-off-by: Rubin Gerritsen <Rubin.Gerritsen@nordicsemi.no>

rugeGerritsen committed:
bluetooth: controller: Add ble_controller_0.3.0-3.prealpha

Adds support for nRF52833 and fixes a bug in hci_data_get().
For more information see the updated CHANGELOG.

Signed-off-by: Rubin Gerritsen <Rubin.Gerritsen@nordicsemi.no>

lemrey committed:
bsdlib: release 0.5.0

* `bsd_irrecoverable_error_handler()` has been removed.
  The application now longer needs to implement it to receive errors
  during initialization, which are instead reported via `bsd_init()`.
* `bsd_shutdown()` now returns an integer.
* Added RAW socket support.
* Added missing AGPS data models.
* Added APGS nofification support.
* Fixed an issue where AGPS data can not be written
  when the GPS socket is in stopped state
* Fixed a memory leak in the GPS socket

Internal revision: 4303c58ecf7

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>
Signed-off-by: Bartosz Gentkowski <bartosz.gentkowski@nordicsemi.no>

rugeGerritsen committed:
bluetooth: controller: Changelog to use :cpp:func: for func references

This will improve the formatting of the documentation

Signed-off-by: Rubin Gerritsen <Rubin.Gerritsen@nordicsemi.no>

sofu1 commented on mpsl/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+                  "(${MPSL_LIB_PATH} doesn't exist.)")
+endif()
+
+set(MPSL_LIB ${MPSL_LIB_PATH}/libmpsl_soft_nrf52.a)
```
remove?

sofu1 replied:
Removed the set(..)

sofu1 commented on mpsl/Kconfig:
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor
```
2019

sofu1 replied:
Fixed

tejlmand committed:
nrf_security: Modified references from nrfxlib to nrf

This commit reworks the way that nrf security reference into nrf docs
due to a current limitation in doxygen setup/doc generation.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

rugeGerritsen committed:
bluetooth: controller: Remove hard-fp and soft-fp from library names

This has been broken for a long time

Signed-off-by: Rubin Gerritsen <Rubin.Gerritsen@nordicsemi.no>
